cmd/core/main.go

```diff
@@ -98,6 +98,8 @@ func main() {
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
 	flag.StringVar(&leaderElectionNamespace, "leader-election-namespace", "",
 		"Determines the namespace in which the leader election configmap will be created.")
+	flag.StringVar(&leaderElectionName, "leader-election-name", "sharding-1",
```

Comment:
1. by default it should be empty.
2. `leader-election-name` -> `app-sharding-key` \c @Somefive  any good suggestion?

Reply:
> 1. by default it should be empty.

I think for a while, if default is empty, and we want to have three sharding, the sharding will contains "", "sharding-1", "sharding-2", it's a little strange. I'm not sure whether I'm proper.

Reply:
After thinking again, most users will only use one sharding. So It is better to default to empty.